lib/charms/traefik_k8s/v0/ingress_per_unit.py

```diff
@@ -14,9 +14,7 @@
 charm's `requirements.txt`.**
 
 ```shell
-cd some-charm
 charmcraft fetch-lib charms.traefik_k8s.v0.ingress_per_unit
```

Comment:
Unless we break the library into two (provides/requires) will charmers not need to add `jsonschema` to their `requirements.txt`?

Reply:
Good point. How does that work for libraries? Do we have a way for libs to declare their dependencies?

Reply:
- How do @jnsgruk  and @mmanciop feel about this? 
![image](https://user-images.githubusercontent.com/6230162/161741495-181c50c3-f471-42b6-a7b7-464f3ce8a116.png)

Reply:
Now that's interesting. I would probably use `log.warning`, but I quite like the idea that it still has a chance of working...

Reply:
either way, you still need to include in the docs to echo jsonschema into requirements.txt or similar :)

Other option is just to validate on the Traefik side, but that will be less reliable.

Reply:
yeah the catch here is that both the provider and the requirer have to install that dependency. Or indeed we could say that Traefik owns the correctness of the data, so the requirer is responsible for doing its own validation if it so desires; and we can provide a utility method or just the public schema for "this is what you should be expecting". Then the charm using the requirer side would have to also install that dependency.